puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -291,7 +291,7 @@ def synthesize_edges(hash, catalog)
                 # case problem here: http://projects.puppetlabs.com/issues/19474
                 # Once that problem is solved and older versions of Puppet that have
                 # the bug are no longer supported we can probably remove this code.
-                unless other_ref =~ /^[A-Z][a-z0-9_-]*(::[A-Z][a-z0-9_-]*)*\[.*\]/m
+                unless other_ref =~ /^[A-Z][a-z0-9_]*(::[A-Z][a-z0-9_]*)*\[.*\]/m
```

Comment:
I think as a separate issue we may want to due an inventory of all of the debt here and figure out what's still valid. Probably risky and time consuming though...